### PR TITLE
(LTH-96) Fix translation of log messages

### PR DESCRIPTION
**Pull request by MikaelSmith**
If no substitution arguments are specified in a log message, translation
wasn't invoked. Ensure that now it will be, so all uses of log messages
behave consistently.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
@parisiale this should be sufficient to fix translating logs.

@briancain I'm surprised you didn't run into this issue. If you do just `LOG_WARNING("some warning");`, it won't actually translate `some warning`. It would only work before if you had a substitution, like `LOG_WARNING("some warning: {1}", 101);`.

**Comment by MikaelSmith** [comment]
@parisiale fixed all your points, just missing testing.

**Comment by briancain** [comment]
@MikaelSmith hmm, that could be because we don't actually use the logging class in leatherman. In our tools we use the translate functions directly.

**Comment by MikaelSmith** [comment]
This is related to [PCP-371](https://tickets.puppetlabs.com/browse/PCP-371).

**Comment by MikaelSmith** [comment]
Added some testing. I'd like to clean up some of the setup between `locale/tests/locale.cc` and `logging/tests/logging_i18n.cc`.

**Comment by MikaelSmith** [comment]
Tests fail everywhere but my laptop :( work for Monday.

**Comment by MikaelSmith** [comment]
Ok, I think this might finally be fixed up.

**Comment by parisiale** [comment]
Tested on OS X; automatic translation for logging::LOG_XXX() calls has been fixed.
Will be testing a bit more.

**Comment by MikaelSmith** [comment]
I'll rebase this in a bit.

File: logging/tests/logging_on_message.cc [hunk 1]
```diff
@@ -1,62 +1,53 @@
...
         LOG_ERROR("error message");
-        THEN("on_message is called with the message") {
-            REQUIRE(level == log_level::error);
-            REQUIRE(message == "error message");
-        }
+        REQUIRE(level == log_level::error);
+        REQUIRE(message == "error message");
     }
-    GIVEN("a FATAL message to log") {
+    SECTION("a FATEL message is logged to on_message") {
```
Comment (parisiale) [inline logging/tests/logging_on_message.cc]: typo

**Comment by MikaelSmith** [comment]
Fixed typo and rebased. I wish I knew why the test disabled in the last commit wasn't working. I'm starting to suspect a bug in Boost 1.58.

**Comment by parisiale** [comment]
Also checked that .mo files are created without running `cmake` twice.
If you're happy to postpone solving the Windows / gcc issue (old Boost?), we can merge this.
👍

**Comment by MikaelSmith** [comment]
I can't reproduce the CI failures (which were avoided with the last commit) at all. Tried setting up an Ubuntu 12.04 instance with essentially the contents of our Travis scripts, and it still passes there.

**Pull request merged**